pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -67,6 +67,10 @@ def __init__(self, pokemon, bot, config):
     ############################################################################
 
     def work(self, response_dict=None):
+        pokeballs = self.bot.item_inventory_count(1)
+        superballs = self.bot.item_inventory_count(2)
+        ultraballs = self.bot.item_inventory_count(3)
+
         response_dict = response_dict or self.create_encounter_api_call()
```

Comment:
shouldn't we **NOT** encounter at all if we don't have any pokeballs. I think the stop check should be above this line.